src/region.js

```diff
@@ -15,18 +15,36 @@ import Slide from './slide';
  */
 function Region({ region }) {
   const [slides, setSlides] = useState([]);
-  const [currentExecutionId, setCurrentExecutionId] = useState(null);
+  const [currentSlide, setCurrentSlide] = useState(null);
+  const [nextSlide, setNextSlide] = useState(null);
+
+  /**
+   * Find the slide after the slide with the fromId.
+   *
+   * @param {number} fromId
+   *   The id from which the next slide is determined.
+   * @returns {JSX.Element}
+   *   The slide.
+   */
+  function findNextSlide(fromId) {
+    const slideIndex = slides.findIndex((slideElement) => slideElement.executionId === fromId);
+    const slide = slides[(slideIndex + 1) % slides.length];
+    return slide;
+  }
 
   /**
    * @param {object} slide
    *   The slide.
    */
   function slideDone(slide) {
     // Go to next slide.
-    setCurrentExecutionId((oldExecutionId) => {
-      const slideIndex = slides.findIndex((slideElement) => slideElement.executionId === oldExecutionId);
-      const nextSlide = slides[(slideIndex + 1) % slides.length];
-      return nextSlide.executionId;
+    setCurrentSlide((oldSlide) => {
```

Comment:
I think my choice of "old" as naming is bad. "previous" would be better. This applies to other places in this file as well.